### ISSUE TITLE
Should we ensure GPU math returns C-equivalent values on domain errors?

### ISSUE THREAD

**Issue by shoaibkamil** [halide/Halide#6404]
(Spinning off from #6394)

We currently test math functions like `pow()` for values outside of the domain, such as `pow(0, -4)`.  The error handling for these functions in C is that they return some specified or implementation-dependent value (depending on which version of the C standard) and set an error signal.  The returned value logic is quite complex, for [pow, there are ~22 cases](https://en.cppreference.com/w/c/numeric/math/pow).  Do we need to require that GPU backends return the correct values on domain/range errors?

**Comment by zvookin** [comment]
Definitely *not* in non-strict-float ("fast math") mode right? Could argue it is the right thing in strict_float. For our own fast implementations, it might be possible to provide consistent results across platforms, etc. but matching the C specification is a pretty big constraint. Moreover, one may want flexibility inside the implementation.

**Comment by shoaibkamil** [comment]
I agree we shouldn't be doing this unless absolutely necessary.  Concretely, I propose removing tests from `correctness_math` that are testing return values for inputs that are out of range or outside of the domain.
